apps/web/src/pages/Pools/PoolAdvertisementPage/components/LanguageRequirementDialog.tsx

```diff
@@ -0,0 +1,486 @@
+      "This position requires that you are bilingual and pass a second language evaluation. The following section on Bilingual positions explains in detail.",
+    id: "X3PC+9",
+    description: "Bilingual advanced language requirement description",
+  },
+  various: {
+    defaultMessage:
+      "Multiple positions are expected to be staffed through this process. You may apply to this process if you meet any of the previous language profiles.",
+    id: "mF1SAO",
+    description: "Various language requirement description",
+  },
+});
+
+type LanguageRequirementDialogProps = {
+  headingLevel?: HeadingProps["level"];
+};
+
+const LanguageRequirementDialog = ({
+  headingLevel = "h3",
+}: LanguageRequirementDialogProps) => {
+  const intl = useIntl();
+  const locale = getLocale(intl);
+
+  return (
+    <Dialog.Root>
+      <Dialog.Trigger>
+        <Button
+          mode="icon_only"
+          color="secondary"
+          icon={InformationCircleIcon}
+          aria-label={intl.formatMessage({
+            defaultMessage: "Learn about how language requirements work.",
+            id: "WhHVMt",
+            description:
+              "Info button label for pool language requirements details.",
+          })}
+        />
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header>
+          {intl.formatMessage({
+            defaultMessage:
+              "Language requirements and second language proficiency",
+            id: "+7yxK0",
+            description: "Heading for the pool language requirements dialog",
+          })}
+        </Dialog.Header>
+        <Dialog.Body>
+          <div
+            data-h2-display="base(grid)"
+            data-h2-gap="base(x1)"
+            data-h2-margin-bottom="base(x1)"
+          >
+            <div>
+              <Heading
+                level={headingLevel}
+                size="h6"
+                data-h2-margin-top="base(0)"
+              >
+                {intl.formatMessage({
+                  defaultMessage: "Language requirements on GC Digital Talent",
+                  id: "h1UJ1Q",
+                  description: "Sub-heading for language requirements dialog",
+                })}
+              </Heading>
+              <p>
+                {intl.formatMessage({
+                  defaultMessage:
+                    "These are the language profiles that you can find on GC Digital Talent",
+                  id: "sry9Wl",
+                  description:
+                    "Language profiles list for language requirements dialog",
+                }) + intl.formatMessage(commonMessages.dividingColon)}
+              </p>
+            </div>
+            <dl data-h2-display="base(grid)" data-h2-gap="base(x.5)">
+              <dt
+                data-h2-color="base(primary.dark)"
+                data-h2-font-weight="base(bold)"
+              >
+                {intl.formatMessage(languageProfiles.englishEssential)}
+              </dt>
+              <dd>
```

Comment:
If we are moving to a description list, it might be better to use the (incorrectly named) `DefinitionList` component to keep things consistent and reduce the need to custom styles,

`import { DefinitionList } from "@gc-digital-talent/ui";`